src/libnetxms/dir.cpp

```diff
@@ -55,8 +55,7 @@ DIR LIBNETXMS_EXPORTABLE *opendir(const char *path)
       return NULL;
 
    // create search pattern
-   if (pattern[len] != '\\')
```

Comment:
This check is actually needed if pattern is C:\ for example.

Reply:
Two cases:
1) pattern[len] != '\\' is true -> set pattern[len] = '\\';
2) pattern[len] != '\\' is false -> already pattern[len] == '\\';
Just keep one line pattern[len] = '\\'; and don't run strange if condition

Reply:
Got it, you are right :)